libraries/AP_GyroFFT/AP_GyroFFT.cpp

```diff
@@ -195,8 +229,7 @@ void AP_GyroFFT::init(uint32_t target_looptime_us)
         const uint16_t loop_rate_hz = 1000*1000UL / target_looptime_us;
         _fft_sampling_rate_hz = loop_rate_hz / _sample_mode;
         for (uint8_t axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
-            _downsampled_gyro_data[axis] = (float*)hal.util->malloc_type(sizeof(float) * _window_size, DSP_MEM_REGION);
-            if (_downsampled_gyro_data[axis] == nullptr) {
+            if (!_downsampled_gyro_data[axis].set_size(_window_size + _samples_per_frame)) {
```

Comment:
Maybe _analysis_enabled should default false, and be set to true when all of this init work passes. Right now it looks as though it will try and do analysis. .... yes, confirmed, if the allocations fail it still tries to do some work

Reply:
ok

Reply:
Fixed